src/views/docs-view/components/docs-root-landing/index.tsx

```diff
@@ -0,0 +1,60 @@
+import { Children, ReactChild } from 'react'
+import LandingHero from 'components/landing-hero'
+import { getElementTextContent } from './utils/get-element-text-content'
+import s from '../../docs-view.module.css'
+
+/**
+ * A structured page view that wraps rendered MDX content,
+ * extracts the first rendered <h1>, and renders a landing hero in its place.
+ *
+ * Used to meet our current design specs for docs landing pages, and to
+ * demonstrate structured page views pattern driven by frontmatter.
+ */
+export default function DocsRootLanding({
+	children,
+	subtitle,
+}: {
+	children: ReactChild[]
+	subtitle: string
+}) {
+	/**
+	 * Iterate over the content children,
+	 * extracting <h1> text to use as the `heading` for our `LandingHero`,
```

Comment:
I briefly considered the option of writing a `remark` plugin to remove the `h1` node, so that we don't have to `Children.map` on each render of content. However, felt like a bit of an optimization, and wanted to get any opinions on whether that would feel worthwhile, so opted not to do it for now.